programs/token-metadata/program/src/state/mod.rs

```diff
@@ -196,11 +196,7 @@ pub trait Resizable: TokenMetadataAccount + BorshSerialize {
             )?;
         }
 
-        let mut account_data = account_info.data.borrow_mut();
-        // passes a slice to borsh so the internal account data array does not get
-        // temporarily resized
-        let mut storage = &mut account_data[..required_size];
-        borsh::to_writer(&mut storage, self)?;
```

Comment:
Sorry I didn't respond until now -- the comment is still valid, since you're still passing a new slice. That's what the `[..]` does, roughly. This does also change the previous logic, since you were writing into a slice that has `required_size` bytes, and now you're using the entire slice of the account. If I were you, I'd revert that bit, but maybe there's something that I'm missing

Reply:
The `required_size` is the size of the account, since the method makes sure the account has the length of `Self::size()`, so it was redundant to have the slice as `[..required_size]`.